crates/core/src/metadata_store.rs

```diff
@@ -299,28 +325,42 @@ impl MetadataStoreClient {
         &self,
         key: ByteString,
     ) -> Result<Option<T>, ReadError> {
-        let value = self.inner.get(key).await?;
-
-        if let Some(mut versioned_value) = value {
-            let value = StorageCodec::decode::<T, _>(&mut versioned_value.value)
-                .map_err(|err| ReadError::Codec(err.into()))?;
-
-            assert_eq!(
-                versioned_value.version,
-                value.version(),
-                "versions must align"
-            );
-
-            Ok(Some(value))
-        } else {
-            Ok(None)
-        }
+        metrics_helper!(
+            @success=METADATA_CLIENT_GET_TOTAL_SUCCESS,
+            @error=METADATA_CLIENT_GET_TOTAL_ERROR,
+            @duration=METADATA_CLIENT_GET_DURATION,
```

Comment:
Separating successes and errors into different metric names makes hard at query time to aggregate. Maybe take a look at an example in task_center.rs that emits the status as dimension instead.
```rust
counter!(TC_FINISHED, "kind" => kind_str, "status" => TC_STATUS_FAILED)
    .increment(1);
```
or from ingress:
```rust
counter!(
    INGRESS_REQUESTS,
    "status" => REQUEST_COMPLETED,
    "rpc.service" => service_name,
    "rpc.method" => handler_name,
)
.increment(1);
````

Reply:
You are right! I should have thought of that :D